triggers/daml/AutoApproval.daml

```diff
@@ -0,0 +1,132 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+
+module AutoApproval where
+
+import Daml.Trigger
+import DA.Record (HasField)
+import Marketplace.Custody.Service qualified as CustodyService
+import Marketplace.Custody.Role qualified as CustodyRole
+import Marketplace.Distribution.Role qualified as DistributionRole
+import Marketplace.Distribution.Auction.Service qualified as AuctionService
+import Marketplace.Distribution.Bidding.Service qualified as BiddingService
+import Marketplace.Issuance.Service qualified as IssuanceService
+import Marketplace.Listing.Service qualified as ListingService
+import Marketplace.Regulator.Service qualified as RegulatorService
+import Marketplace.Settlement.Service qualified as SettlementService
+import Marketplace.Trading.Role qualified as TradingRole
+import Marketplace.Trading.Service qualified as TradingService
+import Marketplace.Trading.Matching.Service qualified as MatchingService
+
+autoApprovalTrigger : Trigger ()
+autoApprovalTrigger = Trigger
+  { initialize = pure ()
+  , updateState = \_ -> pure ()
+  , rule = handleApprovalRule
+  , registeredTemplates = RegisteredTemplates [ registeredTemplate @CustodyRole.Offer
+    , registeredTemplate @CustodyRole.Request
+    , registeredTemplate @CustodyService.Offer
+    , registeredTemplate @DistributionRole.Offer
+    , registeredTemplate @DistributionRole.Request
+    , registeredTemplate @IssuanceService.Offer
+    , registeredTemplate @ListingService.Offer
+    , registeredTemplate @RegulatorService.Offer
+    , registeredTemplate @SettlementService.Offer
+    , registeredTemplate @SettlementService.Request
+    , registeredTemplate @TradingRole.Offer
+    , registeredTemplate @TradingRole.Request
+    , registeredTemplate @MatchingService.Offer
+    , registeredTemplate @MatchingService.Request
+    , registeredTemplate @CustodyService.Request
+    , registeredTemplate @IssuanceService.Request
+    , registeredTemplate @CustodyRole.Role
+    , registeredTemplate @AuctionService.Request
+    , registeredTemplate @BiddingService.Request
+    , registeredTemplate @DistributionRole.Role
+    , registeredTemplate @TradingService.Request
+    , registeredTemplate @ListingService.Request
+    , registeredTemplate @TradingRole.Role ]
+  , heartbeat = None
+  }
+
+-- | Approval all requests
+handleApprovalRule : Party -> TriggerA () ()
+handleApprovalRule party = do
+  debug "Running Auto Approval rule..."
+
+  -- Exercise Accept/Approve choice on the triggered contract
+  custodyRoleOffers         <- generateCommands CustodyRole.Accept        <$> queryByField @CustodyRole.Offer         @"provider" (.provider) (== party)
```

Comment:
Are you not able to omit the type annotation `@provider` here? why isn't it inferred from `(.provider)`?

Reply:
Also, as you're using `(== party)` in every call, I would consider removing that argument, rename `queryFilterParty` or something like that.

Reply:
To your second point, you're correct - I had also through of refactoring this but left it since this trigger is a `catch all` for accept/approval of any contract which we may require other filtering other than `Party` in the future. 

I'll replace this with your suggestion since we currently don't know what these future requirements are and currently we only need to filter on `Party`